### ISSUE TITLE
Detect empty slots

### ISSUE THREAD

**Issue by Irfy** [4rtzel/poe-archnemesis-scanner#53]
I am consistently getting misses on a full inventory and use this one-liner to verify the number of detected slots:

```
python -c "while True: print(len(sum(eval(input()).values(),[])));"
```

I recommend adding a black box to the pictures to be explicitly detected as the most probable match for empty slots. When this is given, the tool can add up the number of detected slots to make sure all 64 slots were detected.

For now, I copy the final output of the scanning every time into the above snippet to make sure I hit the number of items I know are in the inventory. If I have misses I then typically reduce the confidence threshold until I hit everything I have.

Automatic confidence adjustment (="Calibration") can be considered an idea (loop, reducing the confidence threshold until all non-empty slots are detected, down to a predefined absolute minimum), but for starters, an indication like `3x Unknown` in some warning-colored text would be perfect.

**Comment by andr9528** [comment]
Yes please. i've had to lower the confidence slightly before, as it was missing 1 or 2 parts some of the times. I lowered my confidence to 0.92, but still suspect that it missed some.

**Comment by Irfy** [comment]
The coordinates are very helpful for debugging, but for the quick glance/overview, it would be even more helpful to have a final log output like "Found 60 recognized tokens, 3 empty slots and 1 unrecognized token" (or whatever wording you use for the "archnemesis things" instead of "token").

More can definitely be done, but let's defer that until we see if the tool works well enough for now. If misses are still reported, it may make sense to implement two more UI-related things on top of the above:

1. Show empty slots as-if they are an archnemesis token of their own, showing them in some colored text (e.g. **3x Empty slot**) right there with the rest of it, allowing them to be moused-over which would highlight the empty slots. This way the user can visually make sure they are, in fact, empty. This could be a checkbox in settings that is enabled or disabled.

1. When a slot matches neither an archnemesis token, **nor** the empty slot image, that would indicate an unrecognized archnemesis token and the tool could force-select the highest confidence match it got even if it's below the threshold, and/or inform the user with a dialog that this happened and that the user should consider lowering the threshold to the lowest value detected for all slots. Right now, as far as I can tell, if no image matches with high-enough confidence, the tool silently skips that slot (?) -- this is bad, and this approach would fix that. This way, the threshold could be more liberally lowered by default too.

**Comment by 4rtzel** [comment]
I quickly added a message with the information about the number of items found in a new v0.2.2 version. The rest of the changes will have to wait until I stabilize the tool.

**Comment by Irfy** [comment]
On first run it consistently showed all slots empty, but after a restart of the tool it showed everything as expected. The message is sufficient for now. If this works well, no need for the remainder of my suggestions.
